docs/source/intro/installation.rst

```diff
@@ -1,16 +1,892 @@
+   You can check the limit by using ``ulimit -n``.
+
+   .. hint:: This may need to be run every time the system starts up.
+
+It may be worth considering adding some of these commands to your ``~/.bashrc`` file, since some of these settings may reset upon reboot.
+
+.. admonition:: Further Reading
+   :class: seealso title-icon-read-more
+
+   For using WSL as a developer, please see the considerations made in our `wiki-page for developers <https://github.com/aiidateam/aiida-core/wiki/Development-environment#using-windows-subsystem-for-linux-wsl>`_.
+
+.. _intro:install:virtual_environments:
+
+Using virtual environments
+==========================
+
+AiiDA depends on a number of third party python packages, and usually on specific versions of those packages.
+In order not to interfere with third party packages needed by other software on your system, we **strongly** recommend isolating AiiDA in a virtual python environment.
+
+.. admonition:: Additional Information
+   :class: seealso title-icon-read-more
+
+   A very good tutorial on python environments is provided by `realpython.com <https://realpython.com/effective-python-environment>`__.
+
+`venv <https://docs.python.org/3/library/venv.html>`__ is module included directly with python for creating virtual environments.
+To create a virtual environment, in a given directory, run:
+
+.. code-block:: console
+
+   $ python3 -m venv /path/to/new/virtual/environment/aiida
+
+The command to activate the environment is shell specific (see `the documentation <https://docs.python.org/3/library/venv.html#creating-virtual-environments>`__.
+With bash the following command is used:
+
+.. code-block:: console
+
+   $ source /path/to/new/virtual/environment/aiida/bin/activate
+
+To leave or deactivate the environment, simply run:
+
+.. code-block:: console
+
+    (aiida) $ deactivate
+
+.. admonition:: Update install tools
+   :class: tip title-icon-tip
+
+   You may need to install ``pip`` and ``setuptools`` in your virtual environment in case the system or user version of these tools is old
+
+   .. code-block:: console
+
+      (aiida) $ pip install -U setuptools pip
+
+If you have `Conda`_ installed then you can directly create a new environment with ``aiida-core`` and (optionally) Postgres and RabbitMQ installed.
+
+.. code-block:: console
+
+   $ conda create -n aiida -c conda-forge python=3.7 aiida-core aiida-core.services pip
+   $ conda activate
+   $ conda deactivate aiida
+
+
+.. _intro:install:aiida-core:
+
+Installing the aiida-core package
+=================================
+
+.. _Conda: https://anaconda.org/conda-forge/aiida-core
+.. _PyPI: https://pypi.python.org/pypi/aiida-core
+.. _github repository: https://github.com/aiidateam/aiida-core
+
+AiiDA can be installed either from the python package index `PyPI`_, `Conda`_ (both good for general use) or directly from the aiida-core `github repository`_ (good for developers).
```

Comment:
I think I'd prefer to split this into "Install with pip (and venv)" and "Install with conda" instead of mixing the two things like this.

Reply:
Fixed in 9323e4353ec3f0b42b84e7ad518d055016959187.
I haven't split into sections, because that doesn't make sense IMO, since you can also install with pip in a Conda environment. Virtual environments are covered in the section above, I've just clarified that it's installing from anaconda cloud and added the console line command

Reply:
I don't think using conda in combination with `venv` makes sense so I must disagree with you on thi

Reply:
There are limitations to using conda and pip in combination. Especially using conda with `venv` is not possible afaik or at least unwise in so far I think that the current structure is sub-optimal. So, while I disagree with you on this matter, I'm ok with leaving this for now, I don't think that this discussion should block this PR.

As a side note: I'd prefer if you could not resolve conversations that are not obviously resolved. Otherwise it is very difficult for me to follow open points for discussion, because I need to manually unhide all resolved conversations to check the status of the discussion.

Reply:
Well it was resolved IMO 😉, but yeh I understand.

> Especially using conda with venv is not possible afaik

pip has nothing to do with venv though:

1. you use venv, pipenv, or conda to create your virtual environment, then 
2. **within** that virtual environment you use either pip, anaconda cloud (i.e. `conda install`), or git directly to install your python packages.

It would make absolutely no sense to create a venv inside conda, but using pip is absolutely fine (although using anaconda cloud is generally better for handling version dependencies)